### ISSUE TITLE
Configurable estimator similar to boruta_py

### ISSUE THREAD

**Issue by w1nk** [chasedehan/BoostARoota#10]
Would it be possible to make the estimator configurable?  Currently you're requiring xgboost, but I'd prefer to be able to try to use lightgbm (https://github.com/szilard/GBM-perf) as it benchmarks a bit faster.

**Comment by w1nk** [comment]
To follow-up, would you be open to a PR implementing such a thing?

**Comment by chasedehan** [comment]
I have been wanting to do this for a while, but haven't been able to think through the implementation as I would ideally like to be able to implement more estimators.  I'm definitely open to a PR on it - let me know what you are thinking for how to structure it: chasedehan at yahoo dot com

**Comment by chasedehan** [comment]
We are starting to look into the implementation of a configurable estimator.

**Comment by chasedehan** [comment]
Just completed an initial insert of classifiers into BoostARoota. 
 It still needs some testing, but can be found in this branch:

[https://github.com/chasedehan/BoostARoota/tree/insert-clf](https://github.com/chasedehan/BoostARoota/tree/insert-clf)

The code is below for the two methods.  The default is still xgboost, but in the bottom section can specify a learner and pass it in.  Right now, the only way it works is if it is a tree based learner with a .feature_importances_ method attached to it.

`import pandas as pd
import urllib

#################
#Madelon Dataset
train_url = 'http://archive.ics.uci.edu/ml/machine-learning-databases/madelon/MADELON/madelon_train.data'
# download the file
raw_data = urllib.request.urlopen(train_url)
train = pd.read_csv(raw_data, delim_whitespace=True, header=None)
train.columns = ["Var"+str(x) for x in range(len(train.columns))]
labels_url = 'http://archive.ics.uci.edu/ml/machine-learning-databases/madelon/MADELON/madelon_train.labels'
raw_data = urllib.request.urlopen(labels_url)
labels = pd.read_csv(raw_data, delimiter=",", header=None)
labels.columns = ["Y"]


#Original method defaults to xgboost
br = BoostARoota(metric='logloss')

br.fit(train,labels)
len(train.columns)
len(br.keep_vars_)
new_train = br.transform(train)
new_train2 = br.fit_transform(train,labels)


#  Test that its working with any classifier
from sklearn.ensemble import RandomForestClassifier
clf = RandomForestClassifier()

br = BoostARoota(metric=None, clf=clf)
new_train = br.fit_transform(train, labels)`

Once I can do a little more testing I will merge it and push changes to a pypi.  If you are able to look at it and let me know if you see anything.  Thanks, Chase

**Comment by chasedehan** [comment]
Ok, so I'm confident it is backward compatible and that it works appropriately.  I haven't been able to test yet whether the results are as good, but I merged with master and pushed up to PyPI.  I am going to close this issue, if anyone has any issues please let me know.
